src/mcts/search.cc

```diff
@@ -748,7 +745,13 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend() {
     //            in the beginning (and there would be no need for "if
     //            (!is_root_node)"), but that would mean extra mutex lock.
     //            Will revisit that after rethinking locking strategy.
-    if (!is_root_node) node = best_edge.GetOrSpawnNode(/* parent */ node);
+    {
+      while (
+          search_->get_or_spawn_lock_.test_and_set(std::memory_order_acquire))
```

Comment:
Mutex emulated with spinlock is usually slower than native mutex.
(if it is the most efficient way to implement it, mutex would already be implemented like that)